fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum.java

```diff
@@ -78,8 +80,9 @@ public Sum(boolean distinct, boolean alwaysNullable, Expression arg) {
     @Override
     public void checkLegalityBeforeTypeCoercion() {
         DataType argType = child().getDataType();
-        if (((!argType.isNumericType() && !argType.isNullType()) || argType.isOnlyMetricType())) {
-            throw new AnalysisException("sum requires a numeric parameter: " + this.toSql());
+        if ((!argType.isNumericType() && !argType.isBooleanType() && !argType.isNullType())
+                || argType.isOnlyMetricType()) {
+            throw new AnalysisException("sum requires a numeric or boolean parameter: " + this.toSql());
```

Comment:
add regression test in `regression-test/suites/nereids_function_p0/agg_function/agg.groovy`

Reply:
done